### DISCUSSION TITLE
HTTP/2 Negotiations prior knowledge

### DISCUSSION THREAD

**Discussion by tridentsx** [encode/httpx#1548] [Ideas]
Seems like today there is no way of controlling the http/2 negotiation, h2 has three options all supported by RFC https://python-hyper.org/projects/hyper-h2/en/latest/negotiating-http2.html ,It must be possible to pick which option to use especially the prior knowledge since right now this fails when attempting connection to http/2 server that doesn't negotiate.

**Comment by lovelydinosaur** [comment]
Related httpcore PR/discussion... https://github.com/encode/httpcore/pull/269

**Reply by tridentsx** [reply]
I think this should be moved to the issue list, since it is an issue we are not able to use the http/httpcore because its not http2 standards compliant. robot was developed to test telco network and this is absolute must to test 5G telco network functions, for now we have created our own branch since the semantics of the provided pull request wasn't approved.

**Reply by lovelydinosaur** [reply]
> I think this should be moved to the issue list

Agreed, although let's track this under the existing https://github.com/encode/httpx/issues/503

Both the "prior knowledge" and "Upgrade" styles should be covered by that.
